ex2_system/source/system_state/state_task.c

```diff
@@ -173,79 +181,33 @@ SYS_returnState power_switch_uhf(bool *uhf_status) {
     uint8_t is_low_pwr;
     status = HAL_UHF_getLowPwr(&is_low_pwr);
-    if (uhf_status == 0) {
+    if (uhf_status == false) {
       if (is_low_pwr) {
         // send any command to wake it up, e.g.,
         uint8_t scw[12];
-        status = HAL_UHF_getSCW(&scw);
+        status = HAL_UHF_getSCW(scw);
         // Here write any code based on the read scw!
       }
     }
-    if (status != 0) {
-      printf("Switching UHF TRX's power failed.");
+    if (status < 0) {
+      ex2_log("Switching UHF TRX's power failed.");
       // An error occurred. Call this function again?
+      return;
     }
   }
-  return status;
-}
-
-/*
- * Independent functions are considered for easier modification.
- * e.g., We might need to initiate a procedure for a subsystem before
- * disconnecting power.
- */
-SYS_returnState power_switch_stx(bool *stx_status) {
-  int8_t current_status = eps_get_pwr_chnl(STX_PWR_CHNL);
-  if (current_status != *stx_status) {
-    eps_set_pwr_chnl(STX_PWR_CHNL, *stx_status);
-    current_status = eps_get_pwr_chnl(STX_PWR_CHNL);
-  }
-  if (current_status == *stx_status) {
-    return (SYS_returnState)current_status;
-  } else {
-    printf("Switching S-band transmitter's power failed.");
-    return SYS_NO_RESPONSE;
-  }
-}
-
-SYS_returnState power_switch_iris(bool *iris_status) {
-  int8_t current_status = eps_get_pwr_chnl(IRIS_PWR_CHNL);
-  if (current_status != *iris_status) {
-    eps_set_pwr_chnl(IRIS_PWR_CHNL, *iris_status);
-    current_status = eps_get_pwr_chnl(IRIS_PWR_CHNL);
-  }
-  if (current_status == *iris_status) {
-    return (SYS_returnState)current_status;
-  } else {
-    printf("Switching Iris's power failed.");
-    return SYS_NO_RESPONSE;
-  }
 }
 
-SYS_returnState power_switch_dfgm(bool *dfgm_status) {
-  int8_t current_status = eps_get_pwr_chnl(DFGM_PWR_CHNL);
-  if (current_status != *dfgm_status) {
-    eps_set_pwr_chnl(DFGM_PWR_CHNL, *dfgm_status);
-    current_status = eps_get_pwr_chnl(DFGM_PWR_CHNL);
-  }
-  if (current_status == *dfgm_status) {
-    return (SYS_returnState)current_status;
-  } else {
-    printf("Switching DFGM's power failed.");
-    return SYS_NO_RESPONSE;
-  }
-}
 
-SYS_returnState power_switch_adcs(bool *adcs_status) {
-  int8_t current_status = eps_get_pwr_chnl(ADCS_PWR_CHNL);
-  if (current_status != *adcs_status) {
-    eps_set_pwr_chnl(ADCS_PWR_CHNL, *adcs_status);
-    current_status = eps_get_pwr_chnl(ADCS_PWR_CHNL);
+sys_returnstate_e power_switch_sys(const uint8_t channel, const bool target_state) {
+  int8_t current_status = eps_get_pwr_chnl(channel);
+  if (current_status != target_state) {
+    eps_set_pwr_chnl(channel, target_state);
+    current_status = eps_get_pwr_chnl(channel);
   }
-  if (current_status == *adcs_status) {
-    return (SYS_returnState)current_status;
+  if (current_status == target_state) {
+    return (sys_returnstate_e)current_status;
   } else {
-    printf("Switching ADCS's power failed.");
+    ex2_log("Switching system power failed.");
```

Comment:
Won't we wanna know which system, though?

Reply:
If ex2_log is called within a task it will prepend the name of the task. Though I don't know if this is in a task

Reply:
Didn't think of that Robert, I'll leave like this for now since this isn't actually touched in the tests, but hopefully we can make this work.